tests/unit/test_cache.py

```
@@ -0,0 +1,538 @@
+import json
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+from redis.exceptions import RedisError
+
+
+# Helper to create a mock get_redis that returns an async mock
+def create_mock_get_redis(mock_redis: AsyncMock):
+    """Create an async function that returns the mock redis."""
+
+    async def mock_get_redis():
```

Ruff (0.14.8)

Missing return type annotation for private function `mock_get_redis`

(ANN202)